### PR TITLE
28 add function to check missingness

### PR DESCRIPTION
**Pull request by guoyi-smh**
Adds `n_missing` function to check missingness. `n_missing` works on both `vector` and `data.frame` or `data.table` and allows user to specify either to return number of missingness or index. `mi2` is kept as an alias of `n_missing`.

### CODE REVIEW

File: R/utils.R [hunk 1]
```diff
@@ -34,20 +34,94 @@ NULL
...
+#' be returned. For `data.frame` or `data.table` input, the function returns a
+#' vector with length equal to the number of rows of input table and any
+#' missingness in the row will result in a returned value of `TRUE` for the row.
+#'
+#' @return (`logical`)
+#' For `vector` input, a vector of logical values the same length of `x` where
+#' each value represents whether that particular element is missing.
+#' For `data.frame` or `data.table` input, a vector of logical values where
+#' length equals to number of rows of input table and each value represents
+#' whether any element of the row is missing.
```
Comment (vaakesan-SMH) [inline R/utils.R:63]: I'm not sure if this is true? Isn't this outputting a named `character` vector and not a `logical` vector?

**Review by vaakesan-SMH — CHANGES_REQUESTED** [review]
Could you add these two functions to `_pkgdown.yml` so they are added to our `pkgdown` documentation?

Otherwise tested and works well!

File: R/utils.R (hunk 1 above)
Comment (guoyi-smh) [reply R/utils.R:63]: It should be `character` or `logical` depending on the input `index` argument. `character` is missed here and I will add it.

**Comment by vaakesan-SMH** [comment]
Just confirming @guoyi-smh that this function now supersedes `GEMINIpkg::count_missing()`? If so we will need to submit a merge request for `GEMINIpkg` that deprecates that function.

**Comment by guoyi-smh** [comment]
> Just confirming @guoyi-smh that this function now supersedes `GEMINIpkg::count_missing()`? If so we will need to submit a merge request for `GEMINIpkg` that deprecates that function.

Yes I think it is safe to deprecate `GEMINIpkg::count_missing()`.

**Comment by vaakesan-SMH** [comment]
> > Just confirming @guoyi-smh that this function now supersedes `GEMINIpkg::count_missing()`? If so we will need to submit a merge request for `GEMINIpkg` that deprecates that function.
> 
> Yes I think it is safe to deprecate `GEMINIpkg::count_missing()`.

I think we will also need to do the same for `mi2()`. Could you create a merge request on `GEMINIpkg` which deprecates these two and assign me as reviewer to that as well?

**Review by vaakesan-SMH — APPROVED** [review]
Looks good! Thanks for the implementation @guoyi-smh.

Curious how we handle `NULL`, because it can't be added to the `na_strings` vector. We don't necessarily have to make any changes now, just wondering if we might need to extend the functionality at some point.

**Comment by guoyi-smh** [comment]
> Looks good! Thanks for the implementation @guoyi-smh.
> 
> Curious how we handle `NULL`, because it can't be added to the `na_strings` vector. We don't necessarily have to make any changes now, just wondering if we might need to extend the functionality at some point.

Good point. I did think of this but ended up not implementing it in this version. So far I have not encountered any `NULL` in our database. In `lab` table there are some `"NULL"` as strings so the function works on them if the user specify `na_strings = "NULL"`. And it takes some hack to create a dataframe/data.table with `NULL`. So I think we are good for now. But definitely it will be great to add in the future as an improvement.

**Comment by vaakesan-SMH** [comment]
> Good point. I did think of this but ended up not implementing it in this version. So far I have not encountered any `NULL` in our database. In `lab` table there are some `"NULL"` as strings so the function works on them if the user specify `na_strings = "NULL"`. And it takes some hack to create a dataframe/data.table with `NULL`. So I think we are good for now. But definitely it will be great to add in the future as an improvement.

Makes sense to me. Feel free to go ahead and resolve merge conflicts, merge this branch (squash all commits when merging), and then merge your branch on `GEMINIpkg`.

Thanks @guoyi-smh !

**Pull request merged**